docs/pages/interfaces.rst

```diff
@@ -3,8 +3,101 @@
 Interfaces
 ==========
 
-On inheritance
---------------
+General Information
+-------------------
+
+All the non-specific interfaces (e.g. MappableN, BindableN, ApplicativeN) can
+have **Nth** types, at the maximum of three possible types. What does this mean?
+
+:class:`MappableN <returns.interfaces.mappable.MappableN>` interface,
+for example, can have one, two or three possible types. See the example below:
+
+.. code:: python
+
+  >>> from typing import NoReturn
+
+  >>> from returns.interfaces.mappable import MappableN, Mappable1, Mappable2, Mappable3
+
+  >>> one_type: MappableN[int, NoReturn, NoReturn]
+  >>> two_types: MappableN[int, str, NoReturn]
+  >>> three_types: MappableN[int, str, bool]
+  >>> # We have a shortcut for each amount of arguments to reduce the boilerplate
+  >>> one_type: Mappable1[int]
+  >>> two_types: Mappable2[int, str]
+  >>> three_type: Mappable3[int, str, bool]
+
+.. note::
+
+  Useful links before you start here:
+
+  * `Functors, Applicatives, And Monads In Pictures <http://adit.io/posts/2013-04-17-functors,_applicatives,_and_monads_in_pictures.html>`_
+  * `Understanding Functor and Monad With a Bag of Peanuts <https://medium.com/beingprofessional/understanding-functor-and-monad-with-a-bag-of-peanuts-8fa702b3f69e>`_
+  * `Variance of generic types <https://mypy.readthedocs.io/en/latest/generics.html#variance-of-generic-types>`_
+  * `If you know map, I will teach you monads <https://www.youtube.com/watch?v=cB0vpg9-YMQ>`_
+
+Mappable
+--------
+
+:class:`Mappable <returns.interfaces.mappable.MappableN>` interface help us to
+create our own ``Functor`` like :class:`Result <returns.maybe.Maybe>`.
+
+.. code:: python
+
+  >>> from typing import Any, Callable, TypeVar
+
+  >>> from returns.interfaces.mappable import Mappable1
+
+  >>> _FunctorValueType = TypeVar('_FunctorValueType', covariant=True)
+  >>> _NewFunctorValueType = TypeVar('_NewFunctorValueType')
+
+  >>> class MyFunctor(Mappable1[_FunctorValueType]):
+  ...     _inner_value: _FunctorValueType
+  ...
+  ...     def __init__(self, inner_value: _FunctorValueType):
+  ...         self._inner_value = inner_value
+  ...
+  ...     def map(  # This method is required by Mappable
+  ...         self,
+  ...         function: Callable[[_FunctorValueType], _NewFunctorValueType]
+  ...     ) -> 'MyFunctor[_NewFunctorValueType]':
+  ...         new_value = function(self._inner_value)
+  ...         return MyFunctor(new_value)
+  ...
+  ...     def __eq__(self, other: Any) -> bool:  # Required to check the identity law
+  ...         if not isinstance(self, type(other)):
+  ...             return False
+  ...         return self._inner_value == other._inner_value
+
+Laws
+~~~~
+
+To make sure your functor implementation is right, you can apply the
+Functors law on it to test.
+
+1. **Identity Law:** When we pass the identity function to the map method,
```

Comment:
I would suggest to explain **why** do we need this laws in the first place.
Let's show some bad situations when these laws are not respected.

Then, we can show that these laws can save you from yourself. 
Related #457

Reply:
Do you have some bad situations examples? And a link explain why we need the laws?
Currently I don't have this material!

Reply:
I actually don't have an experience with broken monads, but let's think of it right now 😅 
I can only think of `Promise` from `js` that sometimes is very hard to use because of it. Like https://buzzdecafe.github.io/2018/04/10/no-promises-are-not-monads

@astynax can you please help us here? Do you have any bad examples in mind?

Reply:
I also invite @morozzzko to collaborate!

Reply:
Here are some example of why it is bad: https://mmhaskell.com/monads/laws

Reply:
Well, I can clearly see a confusion here. Mainly because I'm confused.

How do we explain **why** an implementation must respect the rule, if the rule is an important part of the definition and the interface? 

I mean, it's just a part of the abstraction. A convention which we use to make it useful. If we want our mappable thing to be useful, it has to follow some rules. Otherwise it just becomes extremely hard to reason about & difficult to work with. 

Let's say we have a mappable which does not respect the identity law. What does it do then? What does it map to? Why does it do that? What's the design benefit? 

I think it's way better not to focus on the **why** and just mention that those lows _make the abstraction work_. 

Perhaps, the docs may ask to send an email if anybody's got a legitimate case when the law must be broken. Would be extremely educational for all parts

Reply:
We can also give an example where `fmap id != id`, because we can create an implementation where `fmap id` produces a side-effect

Reply:
I'll add this code below as an example:
```python
class Number(...):
    def map(self, function: Callable[[_Type], _NewType]) -> 'Number[_NewType]':
        return Number(2 * function(self._inner_value))
```